pkg/webhook/v1beta1/experiment/mutate_webhook.go

```diff
@@ -34,21 +34,13 @@ type ExperimentDefaulter struct {
 }
 
 // NewExperimentDefaulter returns a new Experiment defaulter with the given client.
-func NewExperimentDefaulter(c client.Client) *ExperimentDefaulter {
+func NewExperimentDefaulter(c client.Client, d *admission.Decoder) *ExperimentDefaulter {
 	return &ExperimentDefaulter{
-		client: c,
+		client:  c,
+		decoder: d,
 	}
 }
 
-// ExperimentDefaulter implements admission.DecoderInjector.
-// A decoder will be automatically injected.
-
-// InjectDecoder injects the decoder.
-func (e *ExperimentDefaulter) InjectDecoder(d *admission.Decoder) error {
-	e.decoder = d
-	return nil
-}
-
```

Comment:
The `admission.DecoderInjector` was removed:

https://github.com/kubernetes-sigs/controller-runtime/commit/927c09e54313d6476b5780b7f029f381d777c149#diff-534f3cf4529f988f5e2759c2b588aece0c14420636ac98ec435271c7b888b04f